src/dds/dds-enumerator.h

```diff
@@ -22,6 +21,18 @@ class dds_enumerator
     void init();
 
 private:
-    eprosima::fastdds::dds::DomainParticipant * participant_;
+
+    class DiscoveryDomainParticipantListener
+        : public eprosima::fastdds::dds::DomainParticipantListener
+    {
+        virtual void on_publisher_discovery( eprosima::fastdds::dds::DomainParticipant * participant,
+                                             eprosima::fastrtps::rtps::WriterDiscoveryInfo && info ) override;
+    } _domain_listener;
+
+    eprosima::fastdds::dds::DomainParticipant * _participant;
+    eprosima::fastdds::dds::Subscriber* _subscriber;
+    eprosima::fastdds::dds::Topic* _topic;
+    eprosima::fastdds::dds::DataReader* _reader;
+    eprosima::fastdds::dds::TypeSupport _type;
```

Comment:
This is a smart ptr? I suggest we name it accordingly because right now it's confusing